shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm

```diff
@@ -22,11 +25,9 @@ @implementation FlutterObservatoryPublisher {
 
 #else
 
-@interface FlutterObservatoryPublisher () <NSNetServiceDelegate>
-@end
-
 @implementation FlutterObservatoryPublisher {
-  fml::scoped_nsobject<NSNetService> _netService;
+  DNSServiceRef _dnsServiceRef;
+  bool _dnsServiceRefInitialized;
```

Comment:
Cant this be replaced with a null check on the dns service ref?

Reply:
Done